test/langtools/tools/javac/SuperInit/EarlyAssignmentNoPreview3.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * @test /nodynamiccopyright/
+ * @bug 8334258
+ * @summary Disallow early assignment if FLEXIBLE_CONSTRUCTORS preview feature is not enabled
+ * @compile/fail/ref=EarlyAssignmentNoPreview3.out -XDrawDiagnostics EarlyAssignmentNoPreview3.java
+ */
+public class EarlyAssignmentNoPreview3 {
+
+    Runnable r;
+
+    public EarlyAssignmentNoPreview3() {
+        this(EarlyAssignmentNoPreview3.this.r = () -> System.out.println("hello"));
```

Comment:
Optional suggestion: you could in principle consolidate these tests by overloading the constructor - e.g. you can have
```Foo(int)
Foo(int, int)
Foo(int, int, int)```
And each of those could delegate to `Foo(Runnable)`, with slightly different expressions

Reply:
I tried that, but you only get one "feature disabled" error per file so it doesn't work to put multiple violations in the same file.